Added ARIMA model for stock predction

DarshAgrawal14 opened the pull request:
Processing of dataset to include day ,  month , year and lags to implement better time series analysis and train a ARIMA model with better accuracy

# Related Issues or bug
  - Info about Issue or bug

Fixes: #162 

# Proposed Changes
 - Info about changes Goes here
 
# Additional Info
  - Anything related Issues 
  
# Screenshots

 Original           | Updated
 :--------------------: |:--------------------:
 **original screenshot** | **updated screenshot **|

UppuluriKalyani commented:
@DarshAgrawal14 can you please show me the results and also attach in readme file.

DarshAgrawal14 commented:
![image](https://github.com/user-attachments/assets/8c92d7bc-03be-42bd-8311-0f1df54e810a)
i have added a readme file in the PR itself , link : [ARIMA_Readme](https://github.com/DarshAgrawal14/ML-Nexus/tree/main/Time%20Series%20Analysis/Stock_prediction/ARIMA#readme)
The major part of the project was that i added additional features such as moving averages which i found by learning stock prediction and dividing dates into peices in order to notice the days which is important for both share peice estimation and time series analysis . general time series analysis include taking dataset , creating model with best p , d , q order and other deep learning steps. but the steps i mentioned is what that makes the model better

DarshAgrawal14 commented:
the screen shot shows the RMSE of my model
